tests/PHPUnit/Unit/DataTableTest.php

```diff
@@ -1188,6 +1189,125 @@ public function testMergeSubtablesKeepsMetadata()
         $this->assertEquals('test', $dataTable->getMetadata('additionalMetadata'));
     }
 
+    public function test_sumRowWithLabel_addsNewRowIfTableDoesNotHaveRowWithSameLabel()
+    {
+        $dataTable = new DataTable();
+        $dataTable->addRowsFromSimpleArray([
+            ['label' => 'one', 'nb_visits' => 10],
+            ['label' => 'two', 'nb_visits' => 20],
+            ['label' => 'three', 'nb_visits' => 30],
+        ]);
+
+        $dataTable->sumRowWithLabel('four', ['nb_visits' => 40]);
+
+        $expectedRows = [
+            ['label' => 'one', 'nb_visits' => 10],
+            ['label' => 'two', 'nb_visits' => 20],
+            ['label' => 'three', 'nb_visits' => 30],
+            ['label' => 'four', 'nb_visits' => 40],
```

Comment:
Is the idea here that we are adding a sum row that supposedly sums all the values in the table? I get that for the test the values don't need to actually add up to the summary row, but it'd be good if they did so that it demonstrates what a correct aggregation would do anyway.

```suggestion
        $dataTable->sumRowWithLabel('four', ['nb_visits' => 60]);

        $expectedRows = [
            ['label' => 'one', 'nb_visits' => 10],
            ['label' => 'two', 'nb_visits' => 20],
            ['label' => 'three', 'nb_visits' => 30],
            ['label' => 'four', 'nb_visits' => 60],
```

Reply:
You maybe misunderstood what the method `sumRowWithLabel` is supposed to do. It won't create a summary row, but it will sum the row that is provided to the (existing) row with the provided label in the datatable. So your suggested change, doesn't make too much sense, though.